test/actions/katello/host/update_system_purpose_test.rb

```diff
@@ -0,0 +1,80 @@
+require 'katello_test_helper'
+
+module Katello::Host
+  class UpdateSystemPurposeTest < ActiveSupport::TestCase
+    include Dynflow::Testing
+    include Support::Actions::Fixtures
+    include FactoryBot::Syntax::Methods
+
+    let(:action_class) { ::Actions::Katello::Host::UpdateSystemPurpose }
+
+    def setup
+      User.current = users(:admin)
+      @host = FactoryBot.create(:host, :with_subscription)
+      @service_level = 'Standard'
+      @purpose_role = 'Red Hat Enterprise Linux Server'
+      @purpose_usage = 'Production'
+      @purpose_addons = ['Test Addon']
+      @action = create_action action_class
+    end
+
+    def test_update_system_purpose
+      plan_action @action, @host, @service_level, @purpose_role, @purpose_usage, @purpose_addons
+
+      assert_equal(@host.subscription_facet.service_level, @service_level)
+      assert_equal(@host.subscription_facet.purpose_role, @purpose_role)
+      assert_equal(@host.subscription_facet.purpose_usage, @purpose_usage)
+      assert_equal(@host.subscription_facet.purpose_addons.pluck(:name), [katello_purpose_addons(:addon).name])
+    end
+
+    def test_update_system_purpose_multiple_addons
+      plan_action @action, @host, @service_level, @purpose_role, @purpose_usage, ['Addon One', 'Addon Two']
+
+      assert_equal(@host.subscription_facet.service_level, @service_level)
+      assert_equal(@host.subscription_facet.purpose_role, @purpose_role)
+      assert_equal(@host.subscription_facet.purpose_usage, @purpose_usage)
+      assert_equal(@host.subscription_facet.purpose_addons.count, 2)
+      assert_equal(@host.subscription_facet.purpose_addons.pluck(:name).first, 'Addon One')
+      assert_equal(@host.subscription_facet.purpose_addons.pluck(:name).second, 'Addon Two')
+    end
+
+    def test_update_host_duplicate_purpose
```

Comment:
These first three tests seem very similar to me. You could use ['Addon One', 'Addon Two', 'Addon Two'] and you'd have the same all the same assertions. Let me know if missing something